### PR TITLE
QE: Wait until all valid channels has been synced in CI

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Wait until all valid channels has been synced in CI.
For that purpose, we monitor the file `products.xml` until it exist in the channels folder (`/var/cache/rhn/repodata/#{channel}`).

CI run: https://ci.suse.de/view/Manager/view/Uyuni/job/uyuni-prs-ci-tests/1447/console

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- Cucumber tests were added

- [x] **DONE**

## Links

Ports:
- Manager-4.2 https://github.com/SUSE/spacewalk/pull/18423

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by nodeg — APPROVED** [review]
LGTM. Thanks also to @vandabarata for helping debugging this yesterday and coming up with the `products.xml` file.

**Comment by nodeg** [comment]
We should wait until the PR CI run is finished before merging. I removed the merge candidate tag and will ping Julio myself.

File: testsuite/features/reposync/srv_wait_for_reposync.feature
Comment (vandabarata) [inline testsuite/features/reposync/srv_wait_for_reposync.feature:20]:
```suggestion
    When I wait until all valid channels have been synced
```

File: testsuite/features/step_definitions/command_steps.rb
Comment (vandabarata) [inline testsuite/features/step_definitions/command_steps.rb:396]:
```suggestion
When(/I wait until all valid channels have been synced$/) do
```

File: testsuite/features/step_definitions/command_steps.rb
Comment (vandabarata) [inline testsuite/features/step_definitions/command_steps.rb:387]: maybe we should add just a one line comment mentioning that's the last file to be synced during a reposync? :thinking: just so it's understandable in the future why we're looking for this one and not the `repomd.xml`, which is the file that actually matters but not the last one to sync

**Review by vandabarata — CHANGES_REQUESTED** [review]
just some tiny nitpicks, but thank you for this PR :D

File: testsuite/features/reposync/srv_wait_for_reposync.feature
Comment (nodeg) [reply testsuite/features/reposync/srv_wait_for_reposync.feature:20]: Nice catch! :wink:

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:387]: Agree!

**Comment by nodeg** [comment]
The CI failed in the core stage: 
image

**Comment by nodeg** [comment]
I found the issue. We run on a 15 SP2 minion when running PRs in the CI but we removed all SP2 related code. This has to be adjusted by @jordimassaguerpla. Or we could introduce a interim solution for it and add 15 SP2 to the list

```ruby
raise "Can't build list of reposyncs to leave running" unless ['12-SP4', '12-SP5', '15-SP3', '15-SP4'].include? os_version
```
image

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:387]: > Maybe you want to ckeck, if no `.new` files exists. On a re-sync, the old files stay, but the new generated files are created with this suffix until all files are ready

Great info, thank you!

Comment (nodeg) [reply testsuite/features/step_definitions/command_steps.rb:387]: Do you want to add it to this PR or do we leave it as is for now @srbarrios?

Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:387]: Added

**Review by nodeg — APPROVED** [review]

**Review by vandabarata — APPROVED** [review]

**Comment by nodeg** [comment]
I still see the following in the PR test
image
and am wondering if we should not remove everything SP2 related?

**Comment by srbarrios** [comment]
> I still see the following in the [PR test](https://ci.suse.de/view/Manager/view/Uyuni/job/uyuni-prs-ci-tests/1451) image and am wondering if we should not remove everything SP2 related?

Maybe yes, but that's independent for the PR I made
The goal of the PR is to assure that the channels are completely sync
If something else needs to be fixed, let's do it in a separate PR, to don't mix scopes.

**Comment by nodeg** [comment]
Rebase done. Triggering a new PR test run.

**Comment by nodeg** [comment]
Building the project takes really long today. I guess because of the temp. issues from yesterday.

**Comment by nodeg** [comment]
Rebase done. The PR test looks good so far. We are already in Core - Init Clients: https://ci.suse.de/view/Manager/view/Uyuni/job/uyuni-prs-ci-tests/1459/

**Pull request merged**
